packages/std/src/types.rs

```diff
@@ -67,10 +67,8 @@ pub struct BlockInfo {
 impl BlockInfo {
     /// Returns the block creation time as a Timestamp in nanosecond precision
     pub fn timestamp(&self) -> Timestamp {
-        Timestamp {
-            seconds: self.time,
-            nanos: self.time_nanos,
-        }
+        let nanos_since_epoch = self.time * 1_000_000_000 + self.time_nanos;
```

Comment:
I wouldn't put this math here (with Timestamp internals), but rather:

`Timestamp::from_seconds(self.time).plus_nanos(self.nanos)` (or another constructor you invent)

Reply:
Yeah, right. When I wrote the code `plus_nanos` did not yet exist.

Anyways, I'll leave it for now and change the type of `BlockInfo.time`, such that it disappears.